src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -190,6 +190,18 @@ public AnalyzerConfigOptionsResult GetOptionsForSourcePath(string sourcePath)
             // we would have merged them earlier.
             foreach (var section in _globalConfig.NamedSections)
             {
+                if (!PathUtilities.IsUnixLikePlatform
```

Comment:
Rather than special casing comparison, I think we should consider just normalizing on read. There are other places in editorconfig where we normalize paths (replacing `\\` with `\`). Believe we sohuld consider drive letters as another normalization 

@jjonescz

Reply:
I'm happy to do that. Should I add to the `PathUtilities` ?

Reply:
> I'm happy to do that. Should I add to the `PathUtilities` ?

I think that would be good. And we can add unit tests to `src/Compilers/Core/CodeAnalysisTest/FileSystem/PathUtilitiesTests.cs`.

Reply:
@jjonescz I was thinking on this more, and the problem isn't the casing we're using in file paths but rather that the drive casing provided in the generated `editorconfig` is unreliable. I think we could normalize some of this logic into PathUtilities, add an issue to track down the inconsistent drive letter casing written, and unblock hot reload going forward for now. I tried to dig into where the casing is coming from but it's beyond my depth without some help since there's a lot of moving parts here.

Reply:
We could simplify a bit by something like 

```csharp
if (PathUtilities.HasDriveLetterAndAreEqual(normalizedPath, section.Name))
{
	var sectionWithoutDrive = section.Name[2..];
	var normalizedPathWithoutDrive = normalizedPath[2..];
	if (normalizedPathWithoutDrive.Equals(section.Name, Section.NameComparer))
	{
			//
	}
}
else if (normalizedPath.Equals(section.Name,  Section.NameComparer))
{
	//
}
```

Reply:
> and the problem isn't the casing we're using in file paths but rather that the drive casing provided in the generated `editorconfig` is unreliable

I see, then comparing "drive letter case insensitive" makes sense.

> We could simplify a bit by something like

We could probably have some utility like that. But I don't like the fact that in your example we would rely on implementation detail of `HasDriveLetterAndAreEqual` that the drive letter has length 2; perhaps the utility could give the paths back instead.

Reply:
> I see, then comparing "drive letter case insensitive" makes sense.

I'm not following. The code for reading editorconfig based paths already has logic to normalize paths, particularly the `\\` problem. Why does changing this code to also normalize drive letter paths (to be upper) not also work here?

Reply:
> The code for reading editorconfig based paths already has logic to normalize paths, particularly the `\\` problem.

That's only for editorconfig path vs file path. For example, if you have `/a//b/.editorconfig`, it should apply to file `/a/b/test.cs`. I believe this drive letter casing problem is also for section names inside the .editorconfig, e.g.,

```editorconfig
[c:/a/b.cs]
some = rule
```

should apply to file `C:/a/b.cs`. (Notice the different casing of `C:`.)

So the call above to CollapseWithForwardSlash normalizes the file path. We could normalize the drive letter there as well. But that would not help because the section names are not normalized. I'm not sure if we couldn't normalize them as well; from quick look through the code it doesn't seem we do any normalization for section names currently.

Note that my previous fix for the `\\` problem works because the other side - editorconfig path - was already normalized via a call to `Path.GetDirectoryName` (which already normalizes double slashes).

Reply:
You can see what my `\\` fix was doing in this test:

https://github.com/dotnet/roslyn/blob/b3d9ff7c9dc9e330b24d6087419dffe611a9dd77/src/Compilers/Core/CodeAnalysisTest/Analyzers/AnalyzerConfigTests.cs#L935-L960

If we add a similar test for this PR, the difference should be better apparent.

Reply:
> . We could normalize the drive letter there as well. But that would not help because the section names are not normalized

Do we have a good understanding of which section names represents files or not? If they all represent files then I think we should start normalizing. 

The compiler is designed to essentially normalize file paths on read and then use ordinal comparison after (where practical). Really want to keep this principal true if possible

Reply:
I think we could, because at the point of this we've created a merged global editorconfig (I think). I'm not sure what paths go through parsing an editorconfig. Could also do this during the global merge in https://github.com/dotnet/roslyn/blob/0a0da2d5399c2255bc72e8fa827277830ac5d61f/src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs#L482

Reply:
Alright, I updated the PR to normalize drive casing when parsing. I need to test in vs code but please take a look